Lib/profiling/sampling/sample.py

```diff
@@ -675,10 +679,13 @@ def _validate_collapsed_format_args(args, parser):
 
 def wait_for_process_and_sample(pid, sort_value, args):
     """Sample the process immediately since it has already signaled readiness."""
-    # Set default collapsed filename with subprocess PID if not already set
+    # Set default filename with subprocess PID if not already set
     filename = args.outfile
-    if not filename and args.format == "collapsed":
-        filename = f"collapsed.{pid}.txt"
+    if not filename:
+        if args.format == "collapsed":
+            filename = f"collapsed.{pid}.txt"
+        elif args.format == "gecko":
```

Comment:
ℹ️ **Maintainability**

This line is part of a nested conditional structure that could be simplified. Consider using a dictionary to map formats to filenames, which can make the code more concise and easier to extend in the future.

**Suggested fix:**
```suggestion
        format_to_filename = {
            "collapsed": f"collapsed.{pid}.txt",
            "gecko": f"gecko.{pid}.json"
        }
        filename = filename or format_to_filename.get(args.format)
```